.github/workflows/build.yml

```diff
@@ -8,38 +8,35 @@ jobs:
     name: Run Checks
     strategy:
       matrix:
-        os: [macos-latest, windows-latest, ubuntu-latest]
+        os: [ macos-latest, windows-latest, ubuntu-latest ]
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Set rustup profile
+        run: rustup set profile minimal
+
       - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
+        run: rustup toolchain install stable
 
-      - name: Cache cargo resources
-        uses: Swatinem/rust-cache@v1
+      - name: Cache dependencies
+        uses: actions/cache@v2
         with:
-          sharedKey: deps
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Am i understand right that we reset cache on every Cargo lock change? I am not sure it is necessary.

Reply:
Not reset, I believe it should just create cache entries per os+deps. They would [coexist up to some storage space](https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) (5Gb), then washing out starts for old (least used) ones.

Reply:
@DenisRybas   another thing here: I don't see `Cargo.lock` committed for the moment so that expression evaluates to `<os>-cargo-`.

@vimmerru do you see any issues with pushing one?

Reply:
I never saw Cargo.lock committed for rust libraries. Partial reasoning is here https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries. I suggest to have a cache key based on Cargo.toml content and current date to become obsolete in one months.

Reply:
yes, that is a good point
@DenisRybas  let's update that part according to our offline discussion: link cache key with Cargo.toml and with year-month date (but the latter might be an overhead, we will return to that later)